src/Farmer/Arm/ManagedIdentity.fs

```diff
@@ -0,0 +1,42 @@
+module Farmer.Arm.ManagedIdentity
+
+open Farmer
+open Farmer.ManagedIdentity
+open Farmer.CoreTypes
+
+let userAssignedIdentities = ResourceType ("Microsoft.ManagedIdentity/userAssignedIdentities", "2018-11-30")
+
+//TODO: Remove optionality?
+/// List of resource ID's for the managed identities when a resource is using user assigned identities.
+let Dependencies = function
+    | Some (UserAssigned identities) ->
+        identities
+        |> List.map (fun identity -> identity.ResourceId)
+    | None
+    | Some SystemAssigned ->
+        []
```

Comment:
@ninjarobot should we not add the resource that the system assigned identity belongs to as a dependency? If so - this implies that the System Assigned half should *also* create / store the ResourceId of that resource.